ticket_upgrade_prediction/evaluator.py

```diff
@@ -24,23 +26,72 @@
 from sklearn.model_selection import train_test_split
 
 import mlflow
+from ticket_upgrade_prediction.models import BaseModel
 
 
 @dataclass
 class Metrics:
-    accuracy: float = None
-    roc_auc: float = None
-    precision: float = None
-    recall: float = None
-    f1: float = None
-    pr_auc: float = None
+    accuracy: Optional[float] = None
+    roc_auc: Optional[float] = None
+    precision: Optional[float] = None
+    recall: Optional[float] = None
+    f1: Optional[float] = None
+    pr_auc: Optional[float] = None
+    epoch: Optional[int] = None
 
     def to_dict(self) -> dict:
         return self.__dict__
 
+    @classmethod
+    def from_multiple_metrics(cls, *args):
+        metrics_list = [metric.to_dict() for metric in args]
+        metrics_df = pd.DataFrame(metrics_list).mean()
+        return cls(**metrics_df.to_dict())
```

Comment:
what is this supposed to do? is it for situations where you pass multiple metrics (from kfold) from multiple evalutions and then get mean of them?

Reply:
Yes, we can have for example few `Metrics` objects and want the mean of it.
So now, we can do `Metrics.from_multiple_metrics(metric_obj_1, metric_obj_2, ...)`